src/main/java/com/minecolonies/coremod/colony/interactionhandling/QuestDeliveryInteraction.java

```diff
@@ -59,7 +60,11 @@ public void onServerResponseTriggered(final int responseId, final Player player,
                 {
                     if (((IQuestDeliveryObjective) objective).hasItem(player, colonyQuest) && ((IQuestDeliveryObjective) objective).tryDiscountItem(player, colonyQuest))
                     {
-                        ((IFinalQuestDialogueAnswer) result).applyToQuest(player, data.getColony().getQuestManager().getAvailableOrInProgressQuest(questId));
+                        if (colonyQuest.getCurrentObjectiveInstance() instanceof IDialogueObjectInstance dialogueObjectInstance)
+                        {
+                            dialogueObjectInstance.setHasInteracted(true);
+                        }
+                        ((IFinalQuestDialogueAnswer) result).applyToQuest(player, colonyQuest);
```

Comment:
After interaction we always move on, I dont see why we need this?

Reply:
This is just to indicate that the objective was completed, we don't really check if the dialogue was in fact completed beforehand, but just to ensure that the objectives are completed, this is here.

Reply:
I mean, does this matter? All "objectives in the past" with a smaller index than current stop mattering/are completed.